./pwnlib/lexer.py: Undefined names in __all__

# Pwntools Issue Template

Thanks for contributing to Pwntools!

When reporting an issue, be sure that you are running the latest released version of pwntools (`pip install --upgrade pwntools`).

Please verify that your issue occurs on 64-bit Ubuntu 14.04.  You can use the Dockerfile on `docker.io` for quick testing.

```
$ docker pull pwntools/pwntools:stable
$ docker run -it pwntools/pwntools:stable
```

If possible, provide a proof-of-concept which demonstrates the problem.  Include any binaries or scripts necessary to reproduce the issue, and please include the full debug output via setting the environment variable `PWNLIB_DEBUG=1`.

---

We are trying to [export these variables in `__all__`](https://github.com/Gallopsled/pwntools/blob/dev/pwnlib/lexer.py#L23-L25) but they are not defined.  Are we making promises that we cannot fulfill?  Where are these variables defined?

[flake8](http://flake8.pycqa.org) testing of https://github.com/Gallopsled/pwntools on Python 2.7.15

$ __flake8 . --count --select=F822 --show-source --statistics__
```
./pwnlib/lexer.py:23:1: F822 undefined name 'LlvmLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'CppObjdumpLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'DObjdumpLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'GasLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'CObjdumpLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'ObjdumpLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'NasmLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'NasmObjdumpLexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
./pwnlib/lexer.py:23:1: F822 undefined name 'Ca65Lexer' in __all__
__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
^
9     F822 undefined name 'LlvmLexer' in __all__
```
__F822__ is the "_showstopper_" [flake8](http://flake8.pycqa.org) issue that can halt the runtime with a NameError. This error is different from most other flake8 issues which are merely "style violations" -- useful for readability but they do not effect runtime safety.
* F822: undefined name `name` in `__all__`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -98,6 +98,7 @@ The table below shows which release corresponds to each branch, and what date th
 - [#2542][2542] Decode `_IO_*` flags in `FileStructure` member
 - [#2592][2592] pwnlib.config: Fix customization of `context.timeout`
 - [#2608][2608] Abort on `libcdb file libc.so --unstrip` if eu-unstrip is not installed
+- [#2611][2611] Cleanup `pwnlib.lexer` exports and imports
 
 [2598]: https://github.com/Gallopsled/pwntools/pull/2598
 [2419]: https://github.com/Gallopsled/pwntools/pull/2419
@@ -122,6 +123,7 @@ The table below shows which release corresponds to each branch, and what date th
 [2542]: https://github.com/Gallopsled/pwntools/pull/2542
 [2592]: https://github.com/Gallopsled/pwntools/pull/2592
 [2608]: https://github.com/Gallopsled/pwntools/pull/2608
+[2611]: https://github.com/Gallopsled/pwntools/pull/2611
 
 ## 4.15.0 (`beta`)
 

--- a/pwnlib/lexer.py
+++ b/pwnlib/lexer.py
@@ -2,27 +2,15 @@ from __future__ import division
 
 import re
 
-from pygments.lexer import DelegatingLexer
 from pygments.lexer import RegexLexer
 from pygments.lexer import bygroups
 from pygments.lexer import include
-from pygments.lexer import using
-from pygments.lexers.c_cpp import CLexer
-from pygments.lexers.c_cpp import CppLexer
-from pygments.lexers.d import DLexer
 from pygments.token import Comment
-from pygments.token import Keyword
 from pygments.token import Name
 from pygments.token import Number
-from pygments.token import Operator
-from pygments.token import Other
 from pygments.token import Punctuation
 from pygments.token import String
 from pygments.token import Text
-
-__all__ = ['GasLexer', 'ObjdumpLexer', 'DObjdumpLexer', 'CppObjdumpLexer',
-           'CObjdumpLexer', 'LlvmLexer', 'NasmLexer', 'NasmObjdumpLexer',
-           'Ca65Lexer']
 
 
 class PwntoolsLexer(RegexLexer):
@@ -108,3 +96,14 @@ class PwntoolsLexer(RegexLexer):
             return True
         elif re.match(r'^\.\w+', text, re.M):
             return 0.1
+
+# Aliases for convenience
+GasLexer = PwntoolsLexer
+ObjdumpLexer = PwntoolsLexer
+DObjdumpLexer = PwntoolsLexer
+CppObjdumpLexer = PwntoolsLexer
+CObjdumpLexer = PwntoolsLexer
+LlvmLexer = PwntoolsLexer
+NasmLexer = PwntoolsLexer
+NasmObjdumpLexer = PwntoolsLexer
+Ca65Lexer = PwntoolsLexer

